app/models/admin.rb

```diff
@@ -10,4 +10,18 @@ class Admin < ApplicationRecord
                     uniqueness: { case_sensitive: false },
                     length: { minimum: 6, maximum: 100 },
                     format: { with: URI::MailTo::EMAIL_REGEXP }
+  validates :password, presence: true,
+                       length: { minimum: 8 },
+                       format: { with: %r{[-!$%^&*()_+|~=`{}\[\]:";'<>?,./\w]{8,}} }
+
+  attr_accessor :current_password
+
+  def update_with_password(params, *options)
+    if params[:current_password].present? && params[:password].blank?
```

Comment:
what happens if `current_password` is not valid? Could you check it, pls?
Expected result: `current_password` should be valid for current admin

Reply:
Checked validations. If `current_password` is not valid, we can not chanhe passwo
![Screenshot from 2021-12-08 17-44-34](https://user-images.githubusercontent.com/26040954/145242961-4e3f9a88-a724-4cd2-a185-eb8e89e16e70.png)
rd